### ISSUE TITLE
Repack possible?

### ISSUE THREAD

**Issue by 60fpshacksrock** [Ekey/NFS.BIN.Tool#4]
Would it be possible to add an option to be able to repack the archives?

Edit: I tried it with NFS Carbon GC USA and it extracted the folders but not the actual game files.

**Comment by z4n1t015** [comment]
Yes, I also wanted a tool to put the files back into a dubbing project, but still no success

**Comment by 60fpshacksrock** [comment]
There is a tool called NFS-Disculator that can extract files from the console version.
https://nfs-tools.blogspot.com/2018/09/nfs-disculator-v10-released.html

**Comment by z4n1t015** [comment]
I know this tool, what I wanted was one capable of reimporting files back
into a Dubbing project. Thanks

Em sex., 14 de jun. de 2024 15:50, 60fpshacksrock ***@***.***>
escreveu:

**Comment by 60fpshacksrock** [comment]
I know.

(But what is all of that at the bottom?)

**Comment by ElTioRata** [comment]
My idea was to replace the ENGLISH files on the US releases with another text localization. Too bad that there isn't an repacking option.

**Comment by GeorgeK1ng** [comment]
@ElTioRata Just keep game unpacked and delete BIN files. It should work fine.

**Comment by z4n1t015** [comment]
> My idea was to replace the ENGLISH files on the US releases with another text localization. Unfortunately I didn't found any program with a repacking option :(

I have a definitive solution. Use the "ZDIR.bms" Script in QuickBMS, and use "Reimport2" to reimport the files back into the game

**Comment by GeorgeK1ng** [comment]
@z4n1t015 Can you post mentioned ZDIR.bms here please?

**Comment by 60fpshacksrock** [comment]
> @z4n1t015 Can you post mentioned ZDIR.bms here please?

https://aluigi.altervista.org/quickbms_scripts.php

**Comment by Ekey** [comment]
> Would it be possible to add an option to be able to repack the archives?
> 
> Edit: I tried it with NFS Carbon GC USA and it extracted the folders but not the actual game files.

Yep, it is possible, format is simple.

![peepoThink](https://github.com/user-attachments/assets/b6fd6468-5f98-4249-b0e3-db01bf6905c4)

Why don't I get notifications that there is a discussion here?

**Comment by GeorgeK1ng** [comment]
That would be so cool. I really want to modify language on NFS: U2 / MW / Carbon on Xbox 360. For example Carbon have my language, But console doesn’t support it, so its needed to swap them and repack it back..

**Comment by Ekey** [comment]
[Out](https://github.com/Ekey/NFS.BIN.Tool/tree/main/NFS.Packer)

I recommend to update the unpacker before you use the packer cuz some files were not unpacked properly!

**Comment by Ekey** [comment]
> @Ekey I have just tried extract NFS: Carbon for X360. I have Europe Collectors ISO from https://romsfun.com/download/need-for-speed-carbon-collectors-edition-2-97842, ISO rebuilded with ISO2GOD. Then extracted with Ultra ISO as usual. Then I tried new unpacked and a lot of files are still as __unknown. I quess repacking wont't work correctly, when not all files are correctly extracted.
> 

Files extracted correctly. **ZDIR** files (all) files don't contains real file names only hashes. Folder **__Unknown** created if the real file name is not found from the list and named as **__Unknown/XXXXXXXX** where **XXXXXXXX** is hash of file name and packer works fine with them don't worry about it :)

https://github.com/Ekey/NFS.BIN.Tool/blob/80409ac0cbb83e48c7ca8ecaf0afb26da78c8313/NFS.Packer/NFS.Packer/FileSystem/Package/BinPack.cs#L63

You can test it like: 
- Unpack files
- Repack files
- Unpack repacked files

**Comment by ThatRetailXbox** [comment]
> There is also a tool called NFS-Disculator that can extract files from the console version. https://nfs-tools.blogspot.com/2018/09/nfs-disculator-v10-released.html
> 
> It would be very interesting also if car mods could be converted from PC to console,

This works I done it this way, but his repack tool doesn't work(for me?), so I can edit stuff swap maps ect, but I cant repack the files.

**Comment by TorutheRedFox** [comment]
if all files are correctly extracted it for sure works, you can even make a cursed PC setup where you pack its files into ZZDATA and delete the unpacked versions or put them elsewhere

only files you may have issues with are mod files as mods often don't go through bFile to load files

**Comment by GeorgeK1ng** [comment]
Extract / Repack work just fine even with unknown files. If you wanna do game translation, it require specific distro to make it possible. Thats only issue, some distros simply ignore required changes..

**Comment by TheLastOfLogic** [comment]
> if all files are correctly extracted it for sure works, you can even make a cursed PC setup where you pack its files into ZZDATA and delete the unpacked versions or put them elsewhere
> 
> only files you may have issues with are mod files as mods often don't go through bFile to load files




Hey thank you, must be just the Xbox 360 that can’t play them, super cool though

**Comment by GeorgeK1ng** [comment]
Im talking about X360 exactly, modded language and it works. What are you trying to modify?

**Comment by TheLastOfLogic** [comment]
> Im talking about X360 exactly, modded language and it works. What are you trying to modify?

Oh yeah? I was trying to unpack and repack to do nfs carbon, and I couldn’t get it to load,but I’ll have to check again!

**Comment by GeorgeK1ng** [comment]
Need for Speed Underground 2 I used **Need for Speed - Underground 2 (Europe) (En,Fr,De,It,Nl,Sv).iso** Modified Dutch to Czech
For Most Wanted - **Need for Speed - Most Wanted (Italy).iso** Modified Italian with Czech strings
Need For Speed Carbon - I used my physical copy DVD as it was only one I found which contains language selection screen - Searched binary in expanded files for DUTCH and modified it to CZECH as game is already translated and just needs to be "unlocked". Here you can grab my untouched one https://ufile.io/tm9d7yac

**Comment by ThatRetailXbox** [comment]
Hallo, its been a minute i dont recall what I was trying?, I believe I wanted to unpack and relaxt but not sure?.Verstuurd vanaf mijn iPhoneOp 8 feb 2026 om 3:35 PM heeft George King ***@***.***> het volgende geschreven:﻿GeorgeK1ng left a comment (Ekey/NFS.BIN.Tool#4)
Need for Speed Underground 2 I used Need for Speed - Underground 2 (Europe) (En,Fr,De,It,Nl,Sv).iso Modified Dutch to Czech
For Most Wanted - Need for Speed - Most Wanted (Italy).iso Modified Italian with Czech strings
Need For Speed Carbon - I used my physical copy DVD as it was only one I found which contains language selection screen - Searched binary in expanded files for DUTCH and modified it to CZECH as game is already translated and just needs to be "unlocked". Here you can grab my untouched one https://ufile.io/tm9d7yac
